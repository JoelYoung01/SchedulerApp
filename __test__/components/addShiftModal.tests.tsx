import React, { useEffect } from "react";
import ReactDOM from "react-dom";
import { render, fireEvent } from "@testing-library/react";
