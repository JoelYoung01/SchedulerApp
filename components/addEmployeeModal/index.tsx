export * from "./AddEmployeeModal";
export * from "./AvailiabilityTabs";
export * from "./AvailabilityEditor";
export * from "./EditEmployeeInfo";
export * from "./TabPanel";
