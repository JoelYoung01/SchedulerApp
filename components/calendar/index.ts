export * from "./Calendar";
export * from "./DailyShifts";
export * from "./ShiftCard";
export * from "./ShiftInfo";
export * from "./WeeklyDate";
