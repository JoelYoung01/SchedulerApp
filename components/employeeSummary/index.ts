export * from "./EmployeeSummary";
export * from "./EmployeeCard";
