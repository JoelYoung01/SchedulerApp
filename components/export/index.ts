export * from "./ExportModal";
export * from "./FormatSelect";
