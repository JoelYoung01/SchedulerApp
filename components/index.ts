export * from "./calendar";
export * from "./export";
export * from "./layout";
