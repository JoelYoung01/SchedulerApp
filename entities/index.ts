export * from "./shift";
export * from "./employee";
export * from "./time";
export * from "./types";
export * from "./schedule";
export * from "./color";
