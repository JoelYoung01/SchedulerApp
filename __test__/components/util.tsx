import { expect } from "@jest/globals";
import type { MatcherFunction } from "expect";

import "@testing-library/jest-dom";
import { render, screen, fireEvent, Matcher } from "@testing-library/react";

import React from "react";
import { ExportModal, ExportType } from "../../components";

const toBeHidden: MatcherFunction<[]> = function toBeHidden(actual) {
  if (!(actual instanceof HTMLElement)) {
    throw new Error("actual must be an HTMLElement");
  }
  let current: HTMLElement | null = actual;
  while (current !== null) {
    if (current.style.display === "none") {
      return {
        message: () =>
          "Expected element or parent not to be hidden by `style.display = 'none'`",
        pass: true
      };
    }
    current = current.parentElement;
  }
  return {
    message: () =>
      "Expected element or parent to be hidden by `style.display = 'none'`",
    pass: false
  };
};
expect.extend({ toBeHidden });

declare module "expect" {
  interface AsymmetricMatchers {
    toBeHidden: () => void;
  }
  interface Matchers<R> {
    toBeHidden: () => R;
  }
}

export function checkExportHide(
  name: string,
  element: JSX.Element,
  selector: Matcher
): void {
  test(name, async() => {
    const exportRef: React.RefObject<React.ReactInstance> = React.createRef();
    // Need to add export modal to add printing events
    render(
      <>
        {element}
        <ExportModal
          componentToExport={exportRef}
          defaultValue={ExportType.pdf}
        />
        <div className="printed" data-testid="shown" />
      </>
    );
    expect(screen.getByText(selector)).not.toBeHidden();
    fireEvent(window, new Event("beforeprint"));
    expect(screen.getByText(selector)).toBeHidden();
    expect(screen.getByTestId("shown")).not.toBeHidden();
    fireEvent(window, new Event("afterprint"));
    expect(screen.getByText(selector)).not.toBeHidden();
  });
}
