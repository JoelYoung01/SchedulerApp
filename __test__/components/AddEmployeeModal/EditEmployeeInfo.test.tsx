import { render } from "@testing-library/react";
import { EditEmployeeInfo } from "../../../components";
import { Color } from "../../../entities";

it("Can Render", () => {
  // Arrange
  const dom = render(
    <EditEmployeeInfo
      name={""}
      color={new Color()}
      maxHours={40}
      minHours={0}
      availableColors={[new Color()]}
      setEmployeeName={(name: string) => null}
      setEmployeeColor={(color: Color) => null}
      setEmployeeMaxHours={(maxHours: number) => null}
      setEmployeeMinHours={(minHours: number) => null}
    />
  );

  // Assert
  expect(dom).toMatchSnapshot();
});
