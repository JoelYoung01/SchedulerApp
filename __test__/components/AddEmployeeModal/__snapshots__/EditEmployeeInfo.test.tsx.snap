// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can Render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1ebqi03-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r0:"
              id=":r0:-label"
            >
              Employee Name
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id=":r0:"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root css-1ht9964-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="employee-color-selector"
            >
              Employee Color
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-izo85p-MuiInputBase-root-MuiInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="employee-color-selector employee-color-select"
                class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                id="employee-color-select"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="Green"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r1:"
              id=":r1:-label"
            >
              Minimum Hours per Week
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id=":r1:"
                type="number"
                value="0"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r2:"
              id=":r2:-label"
            >
              Maximum Hours per Week
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id=":r2:"
                type="number"
                value="40"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1ebqi03-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r0:"
            id=":r0:-label"
          >
            Employee Name
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r0:"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root css-1ht9964-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            id="employee-color-selector"
          >
            Employee Color
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-izo85p-MuiInputBase-root-MuiInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="employee-color-selector employee-color-select"
              class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
              id="employee-color-select"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="Green"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r1:"
            id=":r1:-label"
          >
            Minimum Hours per Week
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r1:"
              type="number"
              value="0"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r2:"
            id=":r2:-label"
          >
            Maximum Hours per Week
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r2:"
              type="number"
              value="40"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
